WMF Framework/Remote Notifications/RemoteNotificationsController.swift

```diff
@@ -310,11 +333,23 @@ public enum RemoteNotificationsControllerError: LocalizedError {
     }
 
     @objc public func updateCacheWithCurrentUnreadNotificationsCount() throws {
-        let currentCount = try numberOfUnreadNotifications().intValue
-        let sharedCache = SharedContainerCache(fileName: SharedContainerCacheCommonNames.pushNotificationsCache)
-        var pushCache = sharedCache.loadCache() ?? PushNotificationsCache(settings: .default, notifications: [])
-        pushCache.currentUnreadCount = currentCount
-        sharedCache.saveCache(pushCache)
+        do {
+            let currentCount = try numberOfUnreadNotifications().intValue
+
+            // Same validation before updating cache
+            if currentCount < 0 || currentCount > 1000 {
+                NSLog("Skipping cache update due to unrealistic count: \(currentCount)")
```

Comment:
We use `DDLogError` here
```suggestion
                DDLogError("Skipping cache update due to unrealistic count: \(currentCount)")
```